skt/reporter.py

```diff
@@ -416,11 +416,7 @@ def getjobresults(self):
         return result
 
     def getreport(self):
-        msg = list()
-
-        if self.cfg.get("krelease"):
-            msg.append("result report for kernel %s"
-                       % self.cfg.get("krelease"))
+        msg = ['Hello,\nthanks for your contribution!\n']
```

Comment:
How about this:

    Hello,

    We at Red Hat use the Linux kernel heavily in our products, and really
    appreciate the work you put into your contributions. We'd like to help
    maintain the kernel at the best possible quality, as it would also benefit us.

    That's why Red Hat Kernel CI team runs automatic tests on patch sets posted to
    development maillists. Below you can find results of one such run on a patch
    set you've been involved with.

First of all, it doesn't assume the recipient is a patch author (who most of
the time won't be), but tries to grateful for all contributions (including
reviews). This also states from the get-go where the message is coming from
(not everyone looks at from), and why we're doing this. It makes it clear
we're benefitting from this, which I think is important. I realize my
language can be a bit dry, so I would appreciate your help in making it
softer. However, I think this is the information we need to present at the
beginning of the message.

Reply:
I can't reply to your comment below which GitHub shows as outdated so I'm replying to both here.

ad footer, I assumed most of the time the email will go to a mailing list as well so anybody who has access to it can reply. Explaining resubscription in unsubscribing message is a good idea though! We don't have the un/re-subscribing in action yet but again, I'd avoid hardcoding the email address there. This is an open project and doing so *without the ability to configure it* is another entry barrier for anyone who'd like to use it.

This applies to the introduction too. It reads like a marketing email instead of a test report (I mean, I want to save it for any propagation I might need in the future because I like it that much, but I don't think it has a place in a test report), and it's too long without actually saying anything that is important *to the developer*. I strongly believe that the report should get to the point fast, so the introduction should only contain necessities ("hi, we ran tests on these patches and this is the result").  Besides, who doesn't look at the sender of the message? Together with the subject it's the first thing you see before you even open it.

Reply:
Yes, we absolutely need to get these company-specific messages out of skt. I agree it's another barrier for use by general public. I think this should be a part of configurable message template. Moreover, once again, I think e-mail sending shouldn't be a part of skt at all, but of sktm. For now we'll have to keep it here, though.

Regarding sounding as a marketing message, I agree, but I also think we are representing the company here and if anything goes wrong the company will be responsible, so we should be open and forthcoming about who actually stands behind the test result. OTOH, it's a great opportunity to promote the company as spending extra money and computing power to get the kernel tested, which is good for everyone. If you make an effort for people, you should make it clear and yourself known, or it will not be appreciated (see e.g. the typical struggle of sysadmins being acknowledged for their work). The language of this can be improved, I agree, but let's get something merged and discuss all together and with other kernel developers.

However, I think you're right about saying the most important thing first. So,
let's try to reorder things (although moving it further might help clarity, I
think):

    Hello,

    We at Red Hat use the Linux kernel heavily in our products, and really
    appreciate the work you put into your contributions. Below you can find
    results of automatic tests we ran on a patchset you're involved with.

    We'd like to help maintain the kernel at the best possible quality, as it
    would also benefit us. That's why Red Hat Kernel CI team maintains a
    system running automatic tests on kernel contributions, and shares the
    results, hoping to help developers and maintainers find issues early.

Reply:
> Yes, we absolutely need to get these company-specific messages out of skt. I agree it's another barrier for use by general public. I think this should be a part of configurable message template. Moreover, once again, I think e-mail sending shouldn't be a part of skt at all, but of sktm.

100% agreed. For anyone who wants to use it, having this introduction would make the project a no-go. Until we get the configuration, I don't think we should hardcode any company info. As you say, anything goes wrong and the company is responsible. Now imagine someone unrelated trying to help with testing kernel on their own but screwing up. While in the end it should come up it wasn't us, it still leaves the bad taste and a mess to clean up (and while people will be able to do it with the configuration too, it would be active impersonation and not accident).

Again, if I was a non-RH developer on the receiving end I'd stop reading after the half of the first sentence (TBH I'd do it even as a Red Hatter because it reminds me of a sponsorship spam which, as a developer, I am not interested in). And that would be the better case, since there is certainly a lot of devels with a short fuse who'd complain about us on social media etc. Why are we doing it belongs to an article or a blog post, possibly before we launch it live on upstream mailing lists so it's easily searchable, but definitely not here.

Reply:
Alright, perhaps you're right about putting it away to a blog post or such. We
can give a "read more" link, where things will be explained, then. I still
think we need to clearly state it's Red Hat, but perhaps not explain
everything.

I think our branding could go into skt for a short while, since we're not
promoting it yet, but since we're going live internally only, that can wait
and we can make it an external e-mail template, and return to branding and the
text later. And then, I think it might be easier to not let it in than to come
around to removing it later.

Alright, how about this:

    Hello,

    We appreciate your contributions to the Linux kernel and would like to help
    test them. Below are the results of automatic tests we ran on a patchset
    you're involved with. We hope it will help you find possible issues earlier.

Reply:
"Read more" link at the end sounds good. That way it wouldn't take the attention away from the test report itself, while people who'd be interested in it could check it out.

This introduction sounds pretty good, I think we found the middle ground :) I'll add it shortly.